### PR TITLE
Allow users to register on MC

### PR DESCRIPTION
**Pull request by Ntu2koTrevor**

### CODE REVIEW

**Comment by Ntu2koTrevor** [comment]
@miltontony Ready for review cc @saeedmb8

**Comment by Ntu2koTrevor** [comment]
login form
registration form

**Comment by Ntu2koTrevor** [comment]
login form
reg form

File: mc2/forms.py [hunk 1]
```diff
@@ -11,3 +14,72 @@ class UserSettingsForm(forms.ModelForm):
...
+                required=True,
+                render_value=False,
+                type='password',
+            )
+        ),
+        label=_("Password Confirmation"))
+
+    first_name = forms.CharField(required=False)
+    last_name = forms.CharField(required=False)
+    email = forms.EmailField(required=False)
```
Comment (miltontony) [inline mc2/forms.py]: Email should be required

File: mc2/forms.py [hunk 2]
```diff
@@ -11,3 +14,72 @@ class UserSettingsForm(forms.ModelForm):
...
+    def clean_username(self):
+        try:
+            user_model.objects.get(
+                username__iexact=self.cleaned_data['username'])
+        except user_model.DoesNotExist:
+            return self.cleaned_data['username']
+        raise forms.ValidationError(_('Username already exists.'))
+
+    def clean_confirm_password(self):
+        password = self.cleaned_data.get('password', None)
```
Comment (miltontony) [inline mc2/forms.py]: `.get('password', None)` is the same as `.get('password')`

File: mc2/forms.py [hunk 3]
```diff
@@ -11,3 +14,72 @@ class UserSettingsForm(forms.ModelForm):
     class Meta:
         model = UserSettings
         fields = ('settings_level', )
+
+
+class CreateAccountForm(forms.Form):
```
Comment (miltontony) [inline mc2/forms.py]: Please inherit from [User Creation Form](https://docs.djangoproject.com/en/1.10/topics/auth/default/#django.contrib.auth.forms.UserCreationForm)

File: mc2/views.py [hunk 4]
```diff
@@ -35,6 +41,36 @@ def get_queryset(self):
         return self.get_controllers_queryset(self.request).order_by('name')
 
 
+class CreateAccountView(FormView):
```
Comment (miltontony) [inline mc2/views.py]: Have a look at [django-registration](https://django-registration.readthedocs.io/en/2.2/quickstart.html#quickstart) for an easier way to setup registration.

Comment (miltontony) [inline mc2/views.py]: Old [YAL codebase](https://github.com/praekelt/yal-merge) for example.

**Comment by saeedmb8** [comment]
👍

**Pull request merged**